resources/lib/addon/parser.py

```diff
@@ -74,19 +93,15 @@ def parse_paramstring(paramstring):
     return params
 
 
-def urlencode_params(*args, **kwargs):
+def urlencode_params(query, **kwargs):
```

Comment:
Perhaps remove this altogether and import urlencode from this module, which will resolve to either the shim or direct call?

from resources.lib.addon.parser import urlencode

Reply:
I did have the same thought at first myself - my only concern was that for PY3 we directly import the module. But now that I think about it, it will be in the namespace since we import _from_ so it should work fine.